src/buffer/out/textBuffer.cpp

```diff
@@ -1345,18 +1354,20 @@ void TextBuffer::_PruneHyperlinks()
 // Arguments:
 // - pos - a COORD on the word you are currently on
 // - wordDelimiters - what characters are we considering for the separation of words
-// - lastCharPos - the position of the last nonspace character in the text buffer (to improve performance)
+// - limitOptional - the last "valid" position in the text buffer (to improve performance)
 // Return Value:
 // - true, if successfully updated pos. False, if we are unable to move (usually due to a buffer boundary)
 // - pos - The COORD for the first character on the "word" (inclusive)
-bool TextBuffer::MoveToNextWord(COORD& pos, const std::wstring_view wordDelimiters, COORD lastCharPos) const
+bool TextBuffer::MoveToNextWord(COORD& pos, const std::wstring_view wordDelimiters, std::optional<til::point> limitOptional) const
 {
     // move to the beginning of the next word
     // NOTE: _GetWordEnd...() returns the exclusive position of the "end of the word"
     //       This is also the inclusive start of the next word.
-    auto copy{ _GetWordEndForAccessibility(pos, wordDelimiters, lastCharPos) };
+    const auto bufferSize{ GetSize() };
+    const auto limit{ limitOptional ? *limitOptional : bufferSize.EndExclusive() };

```

Comment:
Isn't there literally a `.value_or` method on `std::optional<T>` for this purpose?

Reply:
Updated w/ `til::coalesce_value()`

Reply:
wait, but if you're just using it with one optional and one value _please prefer `opt.value_or`